src/LinearAlgebra/SparseMatrix/CsrMatrix.hpp

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include <cstdint>
+#include <cstring>
+#include <iostream>
+
+#include <LinearAlgebra/MemoryUtils.hpp>
+
+namespace GridKit
+{
+  namespace LinearAlgebra
+  {
+    using real_type  = double;
+    using index_type = std::int32_t;
+
+    class CsrMatrix
+    {
```

Comment:
It would be good to add Doxygen documentation to the class `CsrMatrix`. We document classes in *.hpp files and class methods in *.cpp files.